### PR TITLE
Add Load Csv Data Example to JDBC examples

### PR DESCRIPTION
**Pull request by kaylaleung**
Updated to be in jdbc directory rather than cloud-client.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:2]:
```suggestion
 * Copyright 2021 Google LLC
```

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:49]: Is the "}" intentional?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:51]: Should these be imported from the library, `com.google.spanner.v1.TypeCode`?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:69]: It seems odd we expect users to implement this theirselves. Is there not a library method?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: How will a user run this sample? Please also see [Method Structure](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#method-structure)

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:261]: Can you use try with resources?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:282]: What should a user do when there is a SQLException?

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (kaylaleung) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: I was under the impression that the example of how users would run the sample would be just to follow the same setup as provided in the test I wrote for this function, but let me know if you mean something else.

Also I'm not exactly sure what kind of main function the Method Structure is referring to as I haven't seen any of the other example files have main functions?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (dttran) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:69]: Not sure what you mean by "users implement this themselves"? Could you elaborate?

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (dttran) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: @averikitsch  I don't think Method Structure applies here, as this is an example similar to other existing examples in the same directory.

**Review by dttran — APPROVED** [review]

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:69]: You expect the user to copy and paste 20+ lines of regex if statements just to set the data type. `parseSpannerDataType` seems like it should be provided by a client library

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: We usually provided an overloaded method, "Any declared function arguments should include a no-arg, main method with examples for how the user can initialize the method arguments and call the entrypoint for the snippet. ". Also does the [README](https://github.com/GoogleCloudPlatform/java-docs-samples/tree/ab0e5d10706468c27dabf27c7b2b20e2c684d8f4/spanner/jdbc) properly describe how to run the snippet?

Comment (kaylaleung) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: Is there an example that you could point me to where this overloaded method is implemented for the other example files? 

Also will update the README 👍🏼

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (kaylaleung) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:69]: I brought this up as a concern with the client team they confirmed that no client library exists that does this and this current method would seem to be the best way.

Comment (averikitsch) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:69]: sg

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:244]: [Example 1 using an overloaded method](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/secretmanager/src/main/java/secretmanager/CreateSecret.java#L28) and [Example 2 using a main method](https://github.com/googleapis/java-bigquery/blob/master/samples/snippets/src/main/java/com/example/bigquery/CopyTable.java#L30)

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:279]: If you use try with resources you shouldn't need to close this. It seems like it would be cleaner to move connection out of the try with resources

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:256]: It may be cleaner not to separately initialize this outside of the CVSParser. The nested tries are harder to follow

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (averikitsch) [inline spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java:222]: You can move this to the top of the file to make it more discoverable

File: spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java
Comment (kaylaleung) [reply spanner/jdbc/src/main/java/com/example/spanner/jdbc/LoadCsvExample.java]: I tried to simplify to 2 tries instead of 3, let me know if that seems cleaner based on my understanding of your comment?

**Review by averikitsch — APPROVED** [review]
Sorry for all the back and forth. This looks great!

**Comment by kaylaleung** [comment]
> Sorry for all the back and forth. This looks great!

Awesome, thank you also for your help :) 

I don't think I have merge access, so I'll leave it to you to merge?

**Pull request merged**
